lib/smart_app_launch/authorization_request_builder.rb

```diff
@@ -0,0 +1,91 @@
+require 'json/jwt'
+
+module SMARTAppLaunch
+  class AuthorizationRequestBuilder
```

Comment:
I would hope that this and `ClientAssertionBuilder` could be combined, as backend services authorization is based on the asymmetric client credentials authorization method. If we need a specific class to handle this for backend services, the class name should reflect that.

Reply:
`ClientAssertionBuilder` and `AuthorizationRequestBuilder` address different scopes and use cases for the tests that use them.  `AuthorizationRequestBuilder`, used for Backend Services, is too broad for use in standalone/EHR tests, and `ClientAssertionBuilder` is too narrow for Backend Services (tests would have duplicated code across them that `AuthRequestBuilder` addresses). 

So I opted to keep both classes but had `AuthRequestBuilder` use `ClientAssertionBuilder` in building the requests, so there isn't redundant behavior between the two classes.  I also renamed `AuthRequestBuilder` to be specific to Backend Services, since those are the only tests that use it.